vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,146 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.HashSet;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+
+    @Value("${page_size}")
+    private int pageSize;
+
+    @RequestMapping(value = "/staff/search")
+    @GetMapping
+    public String getAllSearchedElementsWithoutNum(@RequestParam(name = "searchText") String searchString) {
+        return String.format("redirect:/staff/searchPage?page=%s&searchText=%s", "1", searchString);
```

Comment:
what's the difference between this and the next method? why does it need to be redirected?

Reply:
This is because whenever any string is searched in the search box first getAllSearchedElementsWithoutNum methods gets hitted and it will redirect to pagenumber 1 then getAllSearchedElements method will run continuously with each pagenumber supplied from staffsearch.html.